### PR TITLE
Cleaning and Syncing PropTypes

### PR DESCRIPTION
**Pull request by ramfox**
- Going through each component and making sure the PropTypes cover all the props used in the component. #31
- Starting to create a common propTypes directory (example can be seen in /components/DataTable.js, `fields` is of propType `fieldsProps`. `fieldsProps` was imported from /propTypes/datasetProps. issue #32

### CODE REVIEW

**Comment by ramfox** [comment]
Questionssssss :)

- **DateInput.js**
    *is `onChange` required in this case? function is passed to render, but will it break if `onChange` is undefined? probs, right?*
    *`name` is required, but default doesn't specify*
- **KeyValueForm.js**
    *bunch of commented out code and proptypes, add a TODO remove?*
    *can't tell what `this.props.data` is, also `KeyValueForm` not used anywhere so no clues there.*
    *would love explaination of `this.onChangeKey`, where there is no function declared as such*
- **KeyValueInput.js**
    *not going to take away something you said is required, but why is `onChangeValue` required?*
- **MetadataForm.js**
    *Where is this 'data' and 'validation' coming from, and how can I find out what the shape is?*
    `data` - title, description, theme, keyword, modified, issued, identifier, accessLevel, license, language, landingPage
    `validation` - title, description, 
    *Why is `onChange`, `onCancel` required?*
- **MonthCalendar.js**
    *all these checking `typeof` stuff, can I take out now that `propTypes` is here*
    *would love to know if I did the `isRequired` right*
- **TimePicker.js**
    *also has this type checking stuff*
    *lots of checking stuff, can I remove?*
    *`isRequired` check here for onChange, pretty sure it `isRequried`*
- **ChangeItem.js**
    *what is `data` here?*

File: src/js/components/BarChart.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import React from 'react'
+import React { PropTypes } from 'react'
```
Comment (b5) [inline src/js/components/BarChart.js]: missing comma, should be `import React, { PropTypes } from 'react'`

File: src/js/components/DataTable.js [hunk 2]
```diff
@@ -39,13 +40,13 @@ const DataTable = ({ data, fields, fetching, fetchedAll, error, onLoadMore }) =>
 }
 
 DataTable.propTypes = {
-  fields: PropTypes.array,
-  data: PropTypes.array,
+  data: PropTypes.arrayOf(PropTypes.array),
```
Comment (b5) [inline src/js/components/DataTable.js:43]: This is lovely. Nice addition

File: src/js/components/form/Clock.js [hunk 3]
```diff
@@ -29,7 +29,9 @@ class Clock extends Component {
   // return a up-incrementer
   up (unit) {
     return (e) => {
-      let t = new Date(this.props.value)
+      const value = this.props.value
```
Comment (b5) [inline src/js/components/form/Clock.js]: should favour destructuring here `const { value } = this.props`

File: src/js/components/form/Clock.js [hunk 4]
```diff
@@ -44,16 +46,18 @@ class Clock extends Component {
...
+      t.setMonth(value.getMonth())
+      t.setDate(value.getDate())
       this.handleChange(t, e)
     }
   }
   // return an down-incrementer
   down (unit) {
     return (e) => {
-      let t = new Date(this.props.value)
+      const value = this.props.value
```
Comment (b5) [inline src/js/components/form/Clock.js]: favour destructuring

File: src/js/components/form/ValidInput.js [hunk 5]
```diff
@@ -21,34 +20,34 @@ const ValidInput = (props) => {
...
   value: PropTypes.string.isRequired,
   // placeholder text for an empty field. default: ""
   placeholder: PropTypes.string,
-  // an error message to displacy
-  error: PropTypes.string,
   // weather or not to actually display any passed-in errors
   showError: PropTypes.bool,
-  // change handler func. will be called with (name, value, event)
-  onChange: PropTypes.func.isRequired,
+  // an error message to displacy
```
Comment (b5) [inline src/js/components/form/ValidInput.js]: *display

File: src/js/components/form/ValidLicenseInput.js [hunk 6]
```diff
@@ -23,29 +22,30 @@ const ValidLicenseInput = (props) => {
...
+  className: PropTypes.string,
   // value to display in the field
   value: PropTypes.string.isRequired,
-  // an error message to displacy
-  error: PropTypes.string,
   // weather or not to actually display any passed-in errors
   showError: PropTypes.bool,
-  // change handler func. will be called with (name, value, event)
-  onChange: PropTypes.func.isRequired,
+  // an error message to displacy
```
Comment (b5) [inline src/js/components/form/ValidLicenseInput.js]: *display

File: src/js/components/form/ValidSelect.js [hunk 7]
```diff
@@ -17,31 +16,32 @@ const ValidSelect = (props) => {
...
+  // change handler func. will be called with (name, value, event)
+  onChange: PropTypes.func.isRequired,
 }
 
 ValidSelect.defaultProps = {
   name: undefined,
-  error: undefined,
   showError: true,
-  placeholder: '',
+  error: undefined,
```
Comment (b5) [inline src/js/components/form/ValidSelect.js:42]: can be removed, as the "default" is already `undefined`

**Review by b5 — CHANGES_REQUESTED** [review]
Nice! Just a few pointers to cleanup, then let's get this merged.

**Review by b5 — APPROVED** [review]

**Pull request merged**